### DISCUSSION TITLE
How to use jsonEach function?

### DISCUSSION THREAD

**Discussion by ddxl123** [simolus3/drift#2865] [Q&A]
The table has a field that is a json array, and the field name is abcArr. 
Get the rows that satisfy the condition that all elements in the abcArr array are greater than 1,
and get the rows that satisfy the condition that at least one element in the abcArr array is less than 1.

**Comment by simolus3** [comment]
You can use `EXIST` subqueries for this. First, create a source reflecting all the elements in the array:

```dart
final each = yourTable.abcArr.jsonEach(this);
```

Requiring that all elements are greater than one is equivalent to checking that none is smaller than that, which can be expressed with exist queries:

```dart
final allGreaterThanOne = notExistsQuery(select(each)
  ..where((tbl) => tbl.value.cast<int>().isSmallerOrEqualValue(1)));
```

Filtering rows having an element that is less than one is a direct exists query too:

```dart
final atLeastOneIsLessThanOne = existsQuery(select(each)
  ..where((tbl) => tbl.value.cast<int>().isSmallerThanValue(1)));
```

You can use these in the final query:

```dart
select(yourTable)..where((_) => allGreaterThanOne)
```
